src/EPG.cpp

```diff
@@ -132,13 +129,65 @@ PVR_ERROR EPG::GetEPGForChannel(int channelUid, time_t start, time_t end, kodi::
         }
 
       }
-      broadcast.SetSeriesNumber(XMLUtils::GetIntValue(pListingNode, "season", EPG_TAG_INVALID_SERIES_EPISODE));
-      broadcast.SetEpisodeNumber(XMLUtils::GetIntValue(pListingNode, "episode", EPG_TAG_INVALID_SERIES_EPISODE));
+
+      int season{ EPG_TAG_INVALID_SERIES_EPISODE };
+      int episode{EPG_TAG_INVALID_SERIES_EPISODE};
+      XMLUtils::GetInt(pListingNode, "season", season);
+      XMLUtils::GetInt(pListingNode, "episode", episode);
+      broadcast.SetEpisodeNumber(episode);
       broadcast.SetEpisodePartNumber(EPG_TAG_INVALID_SERIES_EPISODE);
+      // Backend could send epidode only as S00 and parts are not support
+      if (season <= 0 || episode == EPG_TAG_INVALID_SERIES_EPISODE)
+      {        
+        std::regex base_regex("^.*\\([eE][pP](\\d+)(?:/?(\\d+))?\\)");
```

Comment:
Could this be `static` to avoid construction on every function call?

Reply:
Sure.